### PR TITLE
Add CSRF guide to security docs

### PR DESCRIPTION
**Pull request by jameshbarton**
# Description

We have a new CSRF security feature that would benefit from further explanation, so I wrote this.

### CODE REVIEW

File: docs/content/guides/security/csrf/_index.md [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+
+A primary benefit of API Gateways like Gloo Edge is that they provide these weapons at a controlled access point and thus can off-load responsibilities from the application team.
+
+One option for Gloo Edge Enterprise users is to  activate its [Web Application Framework]({{% versioned_link_path fromRoot="/guides/security/waf/" %}}) based on Apache ModSecurity.  It supports use of CSRF rules in the OWASP [Core Rule Set](https://coreruleset.org/).
+
+Alternatively, Envoy provides a simple [CSRF filter](https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/http/csrf/v2/csrf.proto) that may be deployed to an entire Gloo `Gateway` or to individual components of a `VirtualService`.  To understand more about how this filter works in Envoy, we recommend playing in their CSRF [sandbox](https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/csrf).
```
Comment (christian-posta) [inline docs/content/guides/security/csrf/_index.md:19]: by individual "components" of a virtualservice do you mean a route?

Comment (jameshbarton) [reply docs/content/guides/security/csrf/_index.md]: Thanks @christian-posta for the review.  I expanded and clarified the statement you cited in latest commit.

**Comment by christian-posta** [comment]
Seems this may also need a rebase onto master?

**Review by kdorosh — APPROVED** [review]

**Review by kdorosh — APPROVED** [review]

**Pull request merged**